src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs

```diff
@@ -564,6 +584,17 @@ internal List<CompletionResult> GetResultHelper(CompletionContext completionCont
                                 // Handle scenarios such as 'gci | Format-Table @{Label=<tab>' if incomplete parsing of the assignment.
                                 return null;
                             }
+                            else if (lastAst is AssignmentStatementAst assignmentAst2)
+                            {
+                                completionContext.ReplacementIndex = replacementIndex += tokenAtCursor.Text.Length;
```

Comment:
Does `$a = (<tab>` and `$a = @(<tab>` also fall in this code path?

Reply:
This code is hit, but since lastAst is not an AssignmentStatementAst, it falls through.